pkg/autodiscovery/providers/kube_services.go

```diff
@@ -37,6 +37,12 @@ type KubeServiceConfigProvider struct {
 	upToDate bool
 }
 
+// KubeletConfigProvider implements the ConfigProvider interface for the kubelet.
+type KubeEndpointConfigProvider struct {
+	lister   listersv1.EndpointsLister
```

Comment:
We should look for the `ad.datadoghq.com/endpoints.*` annotations on the kube **service** object, not the endpoint itself, as the endpoint object is usually generated by the controller itself. I think we might as well emit templates from `KubeServiceConfigProvider` itself. WDYT?

Reply:
Agree, have a look at the changes please